File: ExampleApp/app/src/main/java/com/motiv/example/DrawerDashboardPresenter.java
package com.motiv.example;

import com.motiv.example.dao.DaoRepository;
import com.motiv.example.dao.LocalStorage;

public class DrawerDashboardPresenter implements DrawerDashboardContract.Presenter {

    private DrawerDashboardContract.View view;
    private GoApi goApi;
    private AuthApi authApi;
    private DaoRepository daoRepository;
    private LocalStorage localStorage;

    public DrawerDashboardPresenter(
            DrawerDashboardContract.View view,
            GoApi goApi,
            AuthApi authApi,
            DaoRepository daoRepository,
            LocalStorage localStorage) {
        this.view = view;
        this.goApi = goApi;
        this.authApi = authApi;
        this.daoRepository = daoRepository;
        this.localStorage = localStorage;
    }

    @Override
    public void eloonClick(android.view.View argument0) {
        view.navigationControllerstartUserActivity(view.getuser());
    }

    @Override
    public void eloonNavigationItemSelected(android.view.MenuItem argument0) {
        view.navigationControllerstartDestinationById(argument0.getItemId());
    }
}

File: ExampleApp/app/src/main/java/com/motiv/example/DrawerDashboardPresenter.kt
 
package com.motiv.example
import androidx.fragment.app.*
import com.motiv.example.dao.DaoRepository
import com.motiv.example.dao.LocalStorage
import dagger.*
import dagger.android.*
import dagger.android.support.*
import javax.inject.*

public class DrawerDashboardPresenter(val view: DrawerDashboardContract.View, val goApi: GoApi, val authApi: AuthApi, val daoRepository: DaoRepository, val localStorage: LocalStorage) : DrawerDashboardContract.Presenter {

    override fun eloonClick(argument0: android.view.View) {
        view.navigationControllerstartUserActivity(view.getuser())
    } override fun eloonNavigationItemSelected(argument0: android.view.MenuItem) {
        view.navigationControllerstartDestinationById(argument0.getItemId())
    }
}
